feat(promtail-mixin): improve standalone usage

Zaunei opened the pull request:
**What this PR does / why we need it**:

This PR improves the usage of the promtail-mixin in setups where no Loki is installed and writes metrics in the same tenant.

With the default Helm installation of promtail, the job label (`job=~"promtail-metrics"`) does not match the currently hard-coded label(`job=~"$namespace/promtail"`), so the latency graphs will not work. Since the selectors will always differ slightly from setup to setup, I made them configurable. 

Added:
- Extend the Makefile to compile promtail-mixins
- Compiled promtail-mixins

Changed:
- Make Dashboard name and label selectors configurable
- Adopt loki-mixin file structure/ Jsonnet stlye

**Which issue(s) this PR fixes**:

No issue created, as I consider this to be a non-breaking fix and improvement.


**Special notes for your reviewer**:

In order not to cause any braking changes I have taken over the values used before. 
**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/loki?pullRequest=8534) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/loki?pullRequest=8534) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/loki?pullRequest=8534) it.</sub>

Zaunei commented:
Found another bug. I could not overwrite the config from the outside, so I fixed the config references and oriented myself to the loki-mixin dashboard generation, also to increase the conciseness here. 
Now I can overwrite the config while importing the mixin.

cstyan commented:
@Zaunei this looks nice :+1: if the fixes here are still relevant to you please update to fix the merge conflicts, and I'll see if we can test deploying this quickly ourselves internally just to confirm nothing's broken

Zaunei commented:
Sure, I will update this PR. In the meantime, ne part of this PR got already upstreamed with #9684.

Zaunei committed:
chore(promtail-mixin): match loki-mixin file structure

Zaunei committed:
feat(promtail-mixin): adjustable name & selectors

Zaunei commented:
@cstyan PR is now uptodate. Happy to hear feedback :)

cstyan commented:
@trevorwhitney does the compiled mixin here make sense?

cstyan commented:
@Zaunei sorry for the delay in response again, Trevor is the one who's done the compiled mixins in the past but unfortunately he's out for the rest of the year AFAIK.

Zaunei commented:
@cstyan @trevorwhitney Is there anything that needs to be improved for this to be merged?

trevorwhitney reviewed (dismissed):
I don't think we _need_ to pre-compile it here but I'm fine either way

Zaunei committed:
feat(makefile): compile promtail-mixin

Zaunei commented:
Thanks for your approval @trevorwhitney!
Since there were merge conflicts and small things were changed by #12527, so I rebased the branch again. I also changed the commit messages directly to conventional commits and dropped the commit for the changelog.md customization. Other than that, I haven't made any changes here and the PR should now be ready again.

JStickler commented:
Closing as stale. Promtail is end of life (EOL) as of March 2, 2026. No future support or updates will be provided. All future feature development will occur in [Grafana Alloy](https://github.com/grafana/alloy).
